runtime/queries/rescript/highlights.scm

```diff
@@ -10,20 +10,28 @@
 [
   (type_identifier)
   (unit_type)
-  "list"
 ] @type
 
+
+(list "list{" @type)
+(list_pattern "list{" @type)
+
+; To ensure that the closing curly bracket is the same color (scope) as the opening curly bracket
+(list "}" @type (#set! "priority" 105))
+(list_pattern "}" @type (#set! "priority" 105))
```

Comment:
We don't implement a `priority` property so the `(#set! ..)`s here can be removed. These blocks can also be combined as long as `list` and `list_pattern` always have `}` nodes:

```scm
(list ["list{" "}"] @type)
(list_pattern ["list{" "}"] @type)
```